doc/OnlineDocs/explanation/analysis/alternative_solutions.rst

```diff
@@ -2,233 +2,359 @@
+arguments are used to limit the search to these solutions:
 
+* ``rel_opt_gap`` : non-negative float or None
 
-Gap Usage Example
------------------
+  * The relative optimality gap for allowable alternative solutions.  None implies that there is no limit on the relative optimality gap (i.e. that any feasible solution can be considered).
 
-When we only want some of the solutions based off a tolerance away from
-optimal, this can be done using the ``abs_opt_gap`` parameter. This is
-shown in the following simple knapsack examples where the weights and
-values are the same.
+* ``abs_opt_gap`` : non-negative float or None
+
+  * The absolute optimality gap for allowable alternative solutions.  None implies that there is no limit on the absolute optimality gap (i.e. that any feasible solution can be considered).
+
+For example, we can generate all optimal solutions as follows:
 
 .. doctest::
    :skipif: not glpk_available
 
-   >>> import pyomo.environ as pyo
-   >>> import pyomo.contrib.alternative_solutions as aos
+   >>> solns = aos.enumerate_binary_solutions(m, num_solutions=100, solver="glpk", abs_opt_gap=0.0)
+   >>> print( [soln.objective().value for soln in solns] )
+   [70.0, 70.0]
 
-   >>> values = [10,9,2,1,1]
-   >>> weights = [10,9,2,1,1]
+Similarly, we can generate the six solutions within 40 of the optimum:
 
-   >>> K = len(values)
-   >>> capacity = 12
+.. doctest::
+   :skipif: not glpk_available
+
+   >>> solns = aos.enumerate_binary_solutions(m, num_solutions=100, solver="glpk", abs_opt_gap=40.0)
+   >>> print( [soln.objective().value for soln in solns] )
+   [70.0, 70.0, 60.0, 60.0, 50.0, 30.0]
 
-   >>> m = pyo.ConcreteModel()
-   >>> m.x = pyo.Var(range(K), within=pyo.Binary)
-   >>> m.o = pyo.Objective(expr=sum(values[i] * m.x[i] for i in range(K)), sense=pyo.maximize)
-   >>> m.c = pyo.Constraint(expr=sum(weights[i] * m.x[i] for i in range(K)) <= capacity)
 
-   >>> solns = aos.enumerate_binary_solutions(m, num_solutions=10, solver="glpk", abs_opt_gap = 0.0)
-   >>> assert(len(solns) == 4)
+Pyomo Solution Pools
+--------------------
 
-In this example, we only get the four ``Solution`` objects that have an
-``objective_value`` of 12.  Note that while we wanted only those four
-solutions with no optimality gap, using a gap of half the smallest value
-(in this case .5) will return the same solutions and avoids any machine
-precision issues.
+The *alternative-solutions library* uses solution pools to filter and store solutions generated by an optimizer.
+The following types of solution pools are currently supported:
+
+* ``keep_all`` : This pool stores all solutions. No solutions are filtered out.
+
+* ``keep_latest`` : This pool stores the latest ``max_pool_size`` solutions that are added to the pool.
+
+  * ``max_pool_size`` (non-negative integer) : The maximum number of solutions that are stored.
+
+* ``keep_latest_unique`` : This pool stores the latest ``max_pool_size`` unique solutions that are added to the pool.
```

Comment:
What's the use case for `keep_latest` rather than `keep_latest_unique`? It seems like having the same solution in the pool repeatedly isn't desirable since they aren't really "alternatives"?